Mathlib/Analysis/Calculus/LocalExtr/Basic.lean

```diff
@@ -83,24 +83,30 @@ theorem posTangentConeAt_mono : Monotone fun s => posTangentConeAt s a := by
   exact ⟨c, d, mem_of_superset hd fun h hn => hst hn, hc, hcd⟩
 #align pos_tangent_cone_at_mono posTangentConeAt_mono
 
-theorem mem_posTangentConeAt_of_segment_subset {s : Set E} {x y : E} (h : segment ℝ x y ⊆ s) :
-    y - x ∈ posTangentConeAt s x := by
-  let c := fun n : ℕ => (2 : ℝ) ^ n
-  let d := fun n : ℕ => (c n)⁻¹ • (y - x)
-  refine ⟨c, d, Filter.univ_mem' fun n => h ?_, tendsto_pow_atTop_atTop_of_one_lt one_lt_two, ?_⟩
-  · show x + d n ∈ segment ℝ x y
-    rw [segment_eq_image']
-    refine ⟨(c n)⁻¹, ⟨?_, ?_⟩, rfl⟩
-    exacts [inv_nonneg.2 (pow_nonneg zero_le_two _), inv_le_one (one_le_pow_of_one_le one_le_two _)]
-  · show Tendsto (fun n => c n • d n) atTop (𝓝 (y - x))
-    exact tendsto_const_nhds.congr fun n ↦ (smul_inv_smul₀ (pow_ne_zero _ two_ne_zero) _).symm
-#align mem_pos_tangent_cone_at_of_segment_subset mem_posTangentConeAt_of_segment_subset
+theorem mem_posTangentConeAt_of_frequently_mem {s : Set E} {x y : E}
+    (h : ∃ᶠ t : ℝ in 𝓝[>] 0, x + t • y ∈ s) : y ∈ posTangentConeAt s x := by
+  obtain ⟨a, ha, has⟩ := Filter.exists_seq_forall_of_frequently h
+  refine ⟨a⁻¹, (a · • y), eventually_of_forall has, tendsto_inv_zero_atTop.comp ha, ?_⟩
+  refine tendsto_const_nhds.congr' ?_
+  filter_upwards [(tendsto_nhdsWithin_iff.1 ha).2] with n (hn : 0 < a n)
+  simp [ne_of_gt hn]
 
 theorem mem_posTangentConeAt_of_segment_subset' {s : Set E} {x y : E}
     (h : segment ℝ x (x + y) ⊆ s) : y ∈ posTangentConeAt s x := by
-  simpa only [add_sub_cancel_left] using mem_posTangentConeAt_of_segment_subset h
+  refine mem_posTangentConeAt_of_frequently_mem (Eventually.frequently ?_)
+  rw [eventually_nhdsWithin_iff]
+  filter_upwards [ge_mem_nhds one_pos] with t ht₁ ht₀
+  apply h
+  rw [segment_eq_image', add_sub_cancel_left]
+  exact mem_image_of_mem _ ⟨le_of_lt ht₀, ht₁⟩
 #align mem_pos_tangent_cone_at_of_segment_subset' mem_posTangentConeAt_of_segment_subset'
 
+theorem mem_posTangentConeAt_of_segment_subset {s : Set E} {x y : E} (h : segment ℝ x y ⊆ s) :
```

Comment:
maybe:
```suggestion
theorem sub_mem_posTangentConeAt_of_segment_subset {s : Set E} {x y : E} (h : segment ℝ x y ⊆ s) :
```
so that you can remove the prime?

Reply:
Asked on [Zulip](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Deprecation.20.26.20recycling.20lemma.20names) about deprecation cycle in this case.